### PR TITLE
MWPW-142224 [MILO][MEP] useBlockCode action improvements

### PR DESCRIPTION
**Pull request by vgoodric**
When using the MEP "useBlockCode" action, the experience columns expect the path to the experimental code.
The easiest way to get the path is to copy the relative path in VS Code.
To reduce user error, MEP will now auto-convert the output into the needed format.
1. convert back slashes to forward slashes for PC users
2. add a forward slash to the beginning of the path

Additionally, the last folder in the path will always be the name of the block and therefore always equal the selector. Rather than introduce a chance for user error, the selector will just be extracted from the path.

Finally, although you can use code from a consumer repo or milo repo, the files can't be found until the 2 repos are merged into one folder. So local development or preview links fail. Code added to use absolute URLs when needed.
Before: https://ubc--homepage--adobecom.hlx.page/homepage/drafts/vgoodrich/fragments/useblockcode/
After: https://ubc--homepage--adobecom.hlx.page/homepage/drafts/vgoodrich/fragments/useblockcode/?milolibs=autoselectorubc

Resolves: [MWPW-142224](https://jira.corp.adobe.com/browse/MWPW-142224)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/autoselectorubc/?martech=off
- After: https://autoselectorubc--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/autoselectorubc/?martech=off

### CODE REVIEW

File: libs/features/personalization/personalization.js [hunk 1]
```diff
@@ -168,7 +168,21 @@ const fetchData = async (url, type = DATA_TYPE.JSON) => {
...
 const consolidateObjects = (arr, prop) => arr.reduce((propMap, item) => {
   item[prop]?.forEach((i) => {
-    propMap[i.selector] = i.val;
+    let { selector, val } = i;
+    if (prop === 'blocks') {
+      if (val?.includes('\\')) val = val?.split('\\').join('/');
+      if (!val?.startsWith('/')) val = `/${val}`;
+      selector = val?.split('/').pop();
+      const { origin } = PAGE_URL;
+      if (origin.includes('.hlx.') || origin.includes('localhost:3000')) {
```
[the remark this answers is not in the document]
Comment (vgoodric) [reply libs/features/personalization/personalization.js:177]: I originally only had localhost but that confused the unit tests.  Options:
1. I check for localhost but exclude the unit test port
2. I update the unit test to expect the output from this if clause
What do you recommend?

File: libs/features/personalization/personalization.js [hunk 2]
```diff
@@ -168,7 +168,21 @@ const fetchData = async (url, type = DATA_TYPE.JSON) => {
 
 const consolidateObjects = (arr, prop) => arr.reduce((propMap, item) => {
   item[prop]?.forEach((i) => {
-    propMap[i.selector] = i.val;
+    let { selector, val } = i;
+    if (prop === 'blocks') {
```
[the remark this answers is not in the document]
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: extracted

File: libs/features/personalization/personalization.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: :3000 removed and unit test updated

File: libs/features/personalization/personalization.js [hunk 3]
```diff
@@ -168,7 +168,21 @@ const fetchData = async (url, type = DATA_TYPE.JSON) => {
...
+      if (val?.includes('\\')) val = val?.split('\\').join('/');
+      if (!val?.startsWith('/')) val = `/${val}`;
+      selector = val?.split('/').pop();
+      const { origin } = PAGE_URL;
+      if (origin.includes('.hlx.') || origin.includes('localhost:3000')) {
+        if (val.startsWith('/libs/')) {
+          val = `${getConfig().miloLibs}${val.replace('/libs', '')}`;
```
[the remark this answers is not in the document]
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: updated

**Review by chrischrischris — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Review by seanchoi-dev — CHANGES_REQUESTED** [review]
Can you resolve
image
?

**Review by seanchoi-dev — APPROVED** [review]

**Pull request merged**